File: kernel/vfs.c
#include "vfs.h"
#include "ramfs.h"
void vfs_init(const uint8_t *disk_image) {
    ramfs_init(disk_image);
}
int vfs_open(const char *name) {
    return ramfs_open(name);
}
uint32_t vfs_size(int fd) {
    return ramfs_size(fd);
}
const uint8_t *vfs_data(int fd) {
    return ramfs_data(fd);
}

File: kernel/vfs.zig
const std: type = @import("std");
extern fn ramfs_init(disk_image: [*]const u8) void;
extern fn ramfs_open(name: [*:0]const u8) c_int;
extern fn ramfs_size(fd: c_int) u32;
extern fn ramfs_data(fd: c_int) [*]const u8;

export fn vfs_init(disk_image: [*]const u8) void {
    ramfs_init(disk_image);
}

export fn vfs_open(name: [*:0]const u8) c_int {
    return ramfs_open(name);
}

//homer lets the barts out  tu tu tu tu

export fn vfs_size(fd: c_int) u32 {
    return ramfs_size(fd);
}

export fn vfs_data(fd: c_int) [*]const u8 {
    return ramfs_data(fd);
}
